lib/resque/job.rb

```diff
@@ -242,6 +242,9 @@ def after_hooks
 
     def failure_hooks
       @failure_hooks ||= Plugin.failure_hooks(payload_class)
```

Comment:
Are there any other places where resque attempts to use `payload_class` and might have the same problem? Would it make sense to rescue the error in `payload_class` and return a dummy job class?

Reply:
That's a great idea, and it beats exceptions/nil checks. I'll give it a shot.

Reply:
Er, now that I think about it... allowing dummy classes to slip in during other payload uses could be surprising, and should fail immediately instead. Notably, `perform` uses payload class to figure out what job to try and run. If a job we're trying to perform is missing, it should raise, not silently use a dummy / null job.

Looking at how payload_class is used, it's mostly internal to Job, and the main external place involving the lookup are during normal work (should raise if missing) and cleanup (should be silently ignored, per this bugfix). So I think we're good for now.